From 713efbc2b62c941e8e7503695c4a23e44194e029 Mon Sep 17 00:00:00 2001
From: Dong Aisheng <b29396@freescale.com>
Date: Wed, 29 Apr 2015 19:15:33 +0800
Subject: [PATCH 0605/5242] MLK-10853-2 dts: imx6ul-ddr3-arm2: add flexcan2
 support

commit  5016edd2eca65f68f410ac16175439c8c6730326 from
https://source.codeaurora.org/external/imx/linux-imx.git

Flexcan2 on ddr3 arm2 board is conflict with uart2, thus a new dts
imx6ul-14x14-ddr3-arm2-flexcan2.dts with uart2 disabled is introduced to
resolve such confliction.

Signed-off-by: Dong Aisheng <b29396@freescale.com>
(cherry picked from commit 14d7595a2f1622c72de1bc534cf18b9e828140ea)

Conflicts:
	arch/arm/boot/dts/Makefile
	arch/arm/boot/dts/imx6ul-ddr3-arm2.dts
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/Makefile                         |    1 +
 .../boot/dts/imx6ul-14x14-ddr3-arm2-flexcan2.dts   |   18 ++++++++++++++++++
 2 files changed, 19 insertions(+)
 create mode 100644 arch/arm/boot/dts/imx6ul-14x14-ddr3-arm2-flexcan2.dts

diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index 3e249ee..85ad5a6 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -569,6 +569,7 @@ dtb-$(CONFIG_SOC_IMX6UL) += \
 	imx6ul-tx6ul-mainboard.dtb \
 	imx6ul-14x14-ddr3-arm2.dtb \
 	imx6ul-14x14-ddr3-arm2-emmc.dtb	\
+	imx6ul-14x14-ddr3-arm2-flexcan2.dtb \
 	imx6ul-14x14-ddr3-arm2-gpmi-weim.dtb \
 	imx6ul-14x14-ddr3-arm2-mqs.dtb \
 	imx6ul-14x14-ddr3-arm2-spdif.dtb \
diff --git a/arch/arm/boot/dts/imx6ul-14x14-ddr3-arm2-flexcan2.dts b/arch/arm/boot/dts/imx6ul-14x14-ddr3-arm2-flexcan2.dts
new file mode 100644
index 0000000..1a3a014
--- /dev/null
+++ b/arch/arm/boot/dts/imx6ul-14x14-ddr3-arm2-flexcan2.dts
@@ -0,0 +1,18 @@
+/*
+ * Copyright (C) 2015 Freescale Semiconductor, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+
+#include "imx6ul-14x14-ddr3-arm2.dts"
+
+&uart2{
+	status = "disabled";
+};
+
+&can2 {
+	status = "okay";
+};
+
-- 
1.7.9.5

